numpy/tests/test_reloading.py

```diff
@@ -12,13 +14,15 @@ def test_numpy_reloading():
     VisibleDeprecationWarning = np.VisibleDeprecationWarning
     ModuleDeprecationWarning = np.ModuleDeprecationWarning
 
-    reload(np)
+    with assert_warns(UserWarning):
```

Comment:
do we want to skip this for pypy ?

Reply:
I dunno, I think it should be fine on PyPy and probably also applies. The sub-interpreter one I do not know/doubt (but its not tested here).

To be honest, I am fine with not doing this at all (and/or at least leaving the sub-interpreter stuff out). But thought I would propose it. It might be a bit too obnoxious.

Reply:
> To be honest, I am fine with not doing this at all (and/or at least leaving the sub-interpreter stuff out). But thought I would propose it. It might be a bit too obnoxious.

Skipping the test sounds fine to me if thats what you mean.

> I dunno, I think it should be fine on PyPy and probably also applies. The sub-interpreter one I do not know/doubt (but its not tested here).

 I tried on pypy, wasn't able to run your ctypes example though since pypy ctypes implementation doesnt have ctypes.pythonapi. But it will probably warn just fine.

Reply:
The ctypes example should do nothing on pypy, since I do not think pypy supports subinterpreters (yet). The reload example should apply just as much to pypy as to cpython (not sure whether it is possible to reload in pypy).

I think this is fairly straight forward, unless there is some unknown failure mode for the sub-interpreter warning being given spuriously. Otherwise mainly the wording matters? I am not even sure how well that ctypes thing works, which is why I did not try to bake it into a test...